### PR TITLE
Firelock Manual Override

### PR DESCRIPTION
**Pull request by 13spacemen**
## About The Pull Request
Alt-click firelocks to open them after 7.5s [as long as you stay still etc.]

Also I renamed the confusing "open permanently" option with the crowbar right-click. Since it did NOT open it permanently. It just forced it open (as opposed to "holding it open")

## Why It's Good For The Game
There are tons of firelocks on station, after a plasma fire they seem to constantly stay closed.
You can open firelocks instantly with the fire alarm but sometimes that too gets destroyed or is out-of-reach of the hallway of firelocks
Having players trapped because they don't have a crowbar is unfun, the game is about social interaction, and given how braindead easy it is to screw up atmos and unleash firelock hell there should be a way to deal with firelock spam for non-powergamers

It is on a 7.5s delay and you must stay still, so if you get stunned or move or whatever, then it gets canceled. If an antag is deliberately trying to kill you 7.5s is not fast enough to outrun them

This change is also great for new players who get trapped in Arrivals because it got spaced by a carp breaking a window

Crowbar is still king for opening firelocks, and after crowbar the fire alarm

https://github.com/tgstation/tgstation/assets/46101244/b5457f78-1239-4a0b-8b9a-1f3bca2b3fdb

## Changelog

:cl:
qol: You can now alt-click firelocks to engage their manual override and open them (takes 7.5 seconds)
/:cl:

### CODE REVIEW

**Comment by Pickle-Coding** [comment]
Just call for help. It's not that hard.

**Comment by bruhlookatthisdood** [comment]
> Just call for help. It's not that hard.

what if
hear me out
no one comes

**Comment by ExcessiveUseOfCobblestone** [comment]
> > Just call for help. It's not that hard.
> 
> what if hear me out no one comes

more likely is tcomms is down and you just dont have a SB radio

File: code/game/machinery/doors/firedoor.dm [hunk 1]
```diff
@@ -550,14 +567,18 @@
 
 /// A simple toggle for firedoors between on and off
 /obj/machinery/door/firedoor/try_to_crowbar_secondary(obj/item/acting_object, mob/user)
+	simple_open_close(user)
+
+/obj/machinery/door/firedoor/proc/simple_open_close(mob/user, manual_opened = FALSE)
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/machinery/doors/firedoor.dm:572]: doc

File: code/game/machinery/doors/firedoor.dm [hunk 2]
```diff
@@ -486,6 +489,20 @@
 			span_warning("You bash [src]!"))
 		playsound(src, bash_sound, 100, TRUE)
 
+/obj/machinery/door/firedoor/AltClick(mob/user)
+	. = ..()
+	if(!user.can_perform_action(src))
+		return
+	try_manual_override(user)
+
+/obj/machinery/door/firedoor/proc/try_manual_override(mob/user)
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/machinery/doors/firedoor.dm:498]: doc

File: code/game/machinery/doors/firedoor.dm [hunk 3]
```diff
@@ -61,6 +61,7 @@
 
 	var/knock_sound = 'sound/effects/glassknock.ogg'
 	var/bash_sound = 'sound/effects/glassbash.ogg'
+	var/manual_open_time = 7.5 SECONDS
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/machinery/doors/firedoor.dm:64]: doc

**Review by ExcessiveUseOfCobblestone — COMMENTED** [review]
[Document](https://github.com/tgstation/tgstation/blob/master/.github/guides/AUTODOC.md) new variables/procs

**Comment by scrunklershrungle932** [comment]
> Having players trapped because they don't have a crowbar is unfun, the game is about social interaction, and given how braindead easy it is to screw up atmos and unleash firelock hell there should be a way to deal with firelock spam for non-powergamers


imo having to get help from other people to open a door seems like it'd involve more social interaction then just using a firelock override

**Comment by 13spacemen** [comment]
@scrunklershrungle932 
That would be fine if it was a one time thing
but right now maps have way too many firelocks spammed everywhere, and one plasma fire is enough to completely screw up the firelocks in a hallway and surrounding areas for the entire round

**Comment by ZephyrTFA** [comment]
just have a crowbar. they are available in every firelocker. Or just ask cargo to print you one

**Pull request closed without merging**